ui/src/shared/components/GaugeChart.js

```diff
@@ -0,0 +1,88 @@
+import React, {PropTypes, Component} from 'react'
+import shallowCompare from 'react-addons-shallow-compare'
+import lastValues from 'shared/parsing/lastValues'
+import Gauge from 'shared/components/Gauge'
+
+import {DEFAULT_COLORS} from 'src/dashboards/constants/gaugeColors'
+import {DASHBOARD_LAYOUT_ROW_HEIGHT} from 'shared/constants'
+
+class GaugeChart extends Component {
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState)
+  }
```

Comment:
We can extend PureComponent now:
https://reactjs.org/docs/react-api.html#reactpurecomponent

Be sure to remove shallowCompare.

Reply:
That's pretty cool, the more you know!